### DISCUSSION TITLE
Autheticating by user rather than datasource.

### DISCUSSION THREAD

**Discussion by mykael22000** [grafana/grafana-infinity-datasource#709] [General]
I've a need to authenticate by each Grafana user, rather than having a single authentication for the DataSource that is shared by all of the Grafana users.  (So one user (Tom) will be able to see some data and another (Harry) will be able to see some different data).

I'm far from being an expert on authentication, but by the looks of the security page, it all seems to be authenticating the server/backend, rather than authenticating each individual Grafana user.  

Any thoughts on how to do this?

**Comment by yesoreyeram** [comment]
What you need is forward oauth identity feature. This will forward grafana auth header to upstream.
